lib/amdResolve.js

```diff
@@ -120,6 +119,24 @@ module.exports = function resolve(pattern, {
                 return acc;
             }, [])
         }, (err, matches) => {
+
+            // determine if the given file path should have the target extension prefix added to it
+            const needsPrefix = (file) => {
+                if(!extensionPrefix) {
+                    return false;
+                }
+                if (file.startsWith(targetExtension) || file.includes(`/${targetExtension}/`)) {
+                    return false;
+                }
+                if (Array.isArray(dependencies) && dependencies.length) {
```

Comment:
these two conditions can be combined since there is no else condition

Reply:
I did it that way for clearer separation of the different logical tests. The more conditions I put into one `if()`, the longer it takes to understand.

Reply:
Yes but we should consider this always as a code smell (imo) or as unnecessary nesting, there is always a way to reduce the amount of the conditions or to invert the logic:

if (!Array.isArray(dependencies) || !dependencies.length) {
    return true;
}
if (dependencies.some(dep => file.startsWith(dep)) ||
     dependencies.some(dep => file.includes(`/${dep}/`))) {
     return false;
}